navigation_and_routing/lib/src/routing/auth_guard.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../auth/auth.dart';
+import 'guard.dart';
+import 'parsed_route.dart';
+
+/// An implementation of [RouteGuard] that redirects to /signIn
+class BookstoreRouteGuard implements RouteGuard<ParsedRoute> {
+  BookstoreAuth auth;
+
+  BookstoreRouteGuard({
+    required this.auth,
+  });
+
+  /// Redirect to /signin if the user isn't signed in.
+  @override
+  Future<ParsedRoute> redirect(ParsedRoute from) async {
```

Comment:
It feels weird to me that  RouteGuard is applied after the route information parser